### DISCUSSION TITLE
How to format date when writing to xlsx

### DISCUSSION THREAD

**Discussion by Curium1** [exceljs/exceljs#1674] [Q&A] [answered]
## 💬 Questions and Help

Hi,
I'm writing to a workbook using workbook.xlsx.write().
My dates are formated like YYYY-MM-DDT00:00:00.000Z in my data due to it being fetched from a SQL server.
I would like to get them formated as a date in Excel.

Using CSV there  seems to be a possibility to format dates using the option property.
workbook.csv.writeFile(filename, options)
Is there something like this for the xlsx.write function?

**Comment by cebbens** [comment, marked as answer]
@Curium1 

You can make use of [styles](https://github.com/exceljs/exceljs#styles) (`numFmt`) when specifying the worksheet columns:
```
ws.columns = [
  { header: 'Id', key: 'id', width: 10 },
  { header: 'Name', key: 'name', width: 32, style: { font: { name: 'Arial Black' } } },
  { header: 'D.O.B.', key: 'DOB', width: 10, style: { numFmt: 'dd/mm/yyyy' } }
]
```

That way, any cell under the DOB column will be styled with that date format.

`numFmt` style prop serves as formatting for numbers, dates and strings. Yeah, really bad naming!

BTW, in most recent version of `exceljs`, it seems that one has to enable styling (defaults to `false`):
```
excel.xlsx.write(stream, {useStyles: true})
excel.xlsx.writeFile(path, {useStyles: true})
```

**Comment by Curium1** [comment]
Thanks!

**Comment by cebbens** [comment]
BTW @Curium1 , don't forget to assign a `Date` object to that column. Otherwise, it won't format it.

**Comment by Curium1** [comment]
I was just about to write the same thing to add to the answer just in case anyone else wondered :)
Once again thank you!

**Comment by cebbens** [comment]
@Curium1 I verified that there is NO need to set `useStyles` to `true` for styling to work, even though it defaults to `false`. That flag may play another role in the lib.

**Comment by vinodhreddygs** [comment]
> @Curium1
> 
> You can make use of [styles](https://github.com/exceljs/exceljs#styles) (`numFmt`) when specifying the worksheet columns:
> 
> ```
> ws.columns = [
>   { header: 'Id', key: 'id', width: 10 },
>   { header: 'Name', key: 'name', width: 32, style: { font: { name: 'Arial Black' } } },
>   { header: 'D.O.B.', key: 'DOB', width: 10, style: { numFmt: 'dd/mm/yyyy' } }
> ]
> ```
> 
> That way, any cell under the DOB column will be styled with that date format.
> 
> `numFmt` style prop serves as formatting for numbers, dates and strings. Yeah, really bad naming!
> 
> BTW, in most recent version of `exceljs`, it seems that one has to enable styling (defaults to `false`):
> 
> ```
> excel.xlsx.write(stream, {useStyles: true})
> excel.xlsx.writeFile(path, {useStyles: true})
> ```

Hi,
Dates are coming as string type, can you please tell me hos to fix? Please refer the issue rinsed i.e. https://github.com/exceljs/exceljs/issues/1666 for reference.
Thanks.

**Comment by Curium1** [comment]
@vinodhreddygs Loop though the array and create a new date for each row of data.
```js
    arrName.forEach((row, index) => {
        return row[column] =  row[column] ? new Date(row[column]).toLocaleDateString('sv-SE') : null
    })
```

**Reply by vinodhreddygs** [reply]
> @vinodhreddygs Loop though the array and create a new date for each row of data.
> 
> ```js
>     arrName.forEach((row, index) => {
>         return row[column] =  row[column] ? new Date(row[column]).toLocaleDateString('sv-SE') : null
>     })
> ```

@Curium1 The actual issue is with cell format it seems. Please look the below issue for  the more details : 
https://github.com/exceljs/exceljs/issues/1666#issuecomment-822898854
